### PR TITLE
Add insights filter docs

### PR DESCRIPTION
**Pull request by felixfbecker**
Adds docs for insights filters (drilldown feature), a background explanation, a how-to, and a quickstart guide step.

@coury-clark @vovakulikov could you in particular please check that all statements are correct? 🙌🏻

### CODE REVIEW

File: doc/code_insights/how-tos/filtering_an_insight.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+| `^github\.com/sourcegraph/go-` | Include or exclude all repositories with the prefix `github.com/sourcegraph/go-` |
+| `service` | Include or exclude all repositories that contain the word `service` in their name |
+| `\.js$` | Include or exclude all repositories that end in `.js` |
+
+You can savely click outside the panel to close it and your filters will still be applied (until a page reload).
+The little dot next to the filters icon indicates that an insight currently has filters applied for you.
+
+### 5. Edit or reset filters
+
+Bring up the filters panel at any time for an insight by clicking the filters button again to edit them or resetting them with the "Reset" buttons.
```
Comment (coury-clark) [inline doc/code_insights/how-tos/filtering_an_insight.md:36]:
```suggestion
Bring up the filters panel at any time for an insight by clicking the filters button again to edit them or reset them with the "Reset" buttons.
```

File: doc/code_insights/how-tos/filtering_an_insight.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+### 3. Update default filters (optional)
+
+By default, the filter will only be visible to you and reset after a page reload.
```
Comment (coury-clark) [inline doc/code_insights/how-tos/filtering_an_insight.md:40]:
```suggestion
By default, the filter will only be visible to you and will reset after a page reload.
```

File: doc/code_insights/how-tos/filtering_an_insight.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+To persist the filter and apply it for other viewers viewing the insight too, click "Update default filters".
+
+### 4. Save as new view (optional)
+
+If you'd like to not modify the insight you're looking at for everyone, but rather fork the insight into a second one with the filters applied for everyone to see an compare to, you can click the button "Save as new view".
```
Comment (coury-clark) [inline doc/code_insights/how-tos/filtering_an_insight.md:45]:
```suggestion
If you'd like to not modify the insight you're looking at for everyone, but rather fork the insight into a second one with the filters applied for everyone to see and compare to, you can click the button "Save as new view".
```

File: doc/code_insights/how-tos/filtering_an_insight.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+The view appears as a separate chart on the dashboard, with default filters applied, while the original insight stays as-is.
```
Comment (coury-clark) [inline doc/code_insights/how-tos/filtering_an_insight.md:46]:
```suggestion
The view appears as a separate chart on the dashboard with default filters applied, while the original insight remains unmodified.
```

**Review by coury-clark — APPROVED** [review]

**Review by Joelkw — APPROVED** [review]
@felixfbecker thank you for kicking these off! I particularly liked the examples chart you included. 

I made some edits and also incorporated @coury-clark 's in-line notes. Feel free to review again but you've got my ok to merge!

File: doc/code_insights/explanations/code_insights_filters.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+# Code Insights filters
+
+Code insights filters allow you to filter a search insight to a subset of repositories. Each search insight has a filter icon that opens the filtering options. 
+
+Filters take effect immediately, without needing to re-process the data series of the insight.
+
+> Note: filters are not yet available on language statistics insights.
```
Comment (felixfbecker) [inline doc/code_insights/explanations/code_insights_filters.md:7]: For alert style
```suggestion
> NOTE: filters are not yet available on language statistics insights. 
```

File: doc/code_insights/explanations/code_insights_filters.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+Code insights filters allow you to filter a search insight to a subset of repositories. Each search insight has a filter icon that opens the filtering options. 
+
+Filters take effect immediately, without needing to re-process the data series of the insight.
+
+> NOTE: filters are not yet available on language statistics insights. 
+
+## Filter options
+
+### Repo: filters
```
Comment (felixfbecker) [inline doc/code_insights/explanations/code_insights_filters.md:11]:
```suggestion
### `repo:` filters
```

File: doc/code_insights/how-tos/filtering_an_insight.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+# Filtering a code insight
+
+This how-to assumes that you already have [created some search insights](../quickstart.md).
+
+> Note: filters are not yet available on language statistics insights.
```
Comment (felixfbecker) [inline doc/code_insights/how-tos/filtering_an_insight.md:5]:
```suggestion
> NOTE: filters are not yet available on language statistics insights. 
```

File: doc/code_insights/explanations/code_insights_filters.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+### Other filtering options
+
+We're currently exploring additional filters that would be valuable. If you have feedback about a particular filter you'd like for code insights, we would [love to hear your feedback](mailto:feedback@sourcegraph.com).
+
+## Filter persistance and sharing
+
+### Filters are temporary by default
+
+By default, filters are temporary (present until you refresh the page) and local (no one else can see them). You can modify filters without affecting what others see, unless or until you save the filters.
```
Comment (vovakulikov) [inline doc/code_insights/explanations/code_insights_filters.md:29]: Maybe is it worth mentioning that if you change the dashboard via dashboard select it also will be lost.

File: doc/code_insights/explanations/code_insights_filters.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+### Saving filters as defaults
+
+You can set filters to be persistent, or default, even after a page reload on a code insights in two ways:
+
+1. Create a filter and click "save/update default filters": this will save the filters so they persist for all viewers of this insight, on any dashboard page the insight appears. 
+1. Create a filter and "save as new view": this will create a new insight chart with your filter applied as the default filter on that insight. It will _not_ also save these filters to the existing insight unless you also select "save/update default filters". 
+
+### Filter indicator 
+
+Insights that have any filters applied will have a small dot on their filter icon. The filter can always be reset by opening the filter panel again and clicking "Reset" above each filter or "Reset all filters" – but note that, like any filter edit, you must also then "update default filters" to save that reset state if you want it to persist.
```
Comment (vovakulikov) [inline doc/code_insights/explanations/code_insights_filters.md:40]: At the moment according to [the latest design sketches](https://www.figma.com/file/1zvF8Owrh1ndVH29lvyrXn/RFC-419%3A-Code-Insights-“Drilldown”?node-id=246%3A11) we don't have the `Reset all filters` button. @AlicjaSuska should we bring it back? I remember we had that button but at some point, it was removed from the drill-down filters panel.

**Review by vovakulikov — APPROVED** [review]
@felixfbecker cool documentation, simple and easy to follow.

File: doc/code_insights/explanations/code_insights_filters.md (hunk 8 above)
Comment (felixfbecker) [reply doc/code_insights/explanations/code_insights_filters.md:29]:
```suggestion
By default, filters are temporary (present until you refresh the page or switch to a different dashboard) and local (no one else can see them). You can modify filters without affecting what others see, unless or until you save the filters.
```

File: doc/code_insights/explanations/code_insights_filters.md (hunk 9 above)
Comment (felixfbecker) [reply doc/code_insights/explanations/code_insights_filters.md:40]:
```suggestion
Insights that have any filters applied will have a small dot on their filter icon. The filter can always be reset by opening the filter panel again and clicking "Reset" above each filter – but note that, like any filter edit, you must also then "update default filters" to save that reset state if you want it to persist. 
```

**Pull request merged**